src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java

```diff
@@ -218,14 +226,15 @@ static ConfigDef newConfigDef()
       .define(BUFFER_FLUSH_TIME_SEC,
         Type.LONG,
         BUFFER_FLUSH_TIME_SEC_DEFAULT,
+        ConfigDef.Range.atLeast(BUFFER_FLUSH_TIME_SEC_MIN),
         Importance.LOW,
         "The time in seconds to flush cached data",
         CONNECTOR_CONFIG,
         3,
         ConfigDef.Width.NONE,
         BUFFER_FLUSH_TIME_SEC)
       .define(SNOWFLAKE_METADATA_ALL,
-        Type.STRING,
+        Type.BOOLEAN,
```

Comment:
is this a behavior change? What does the config file look like after this change?

Reply:
I didn't modify the test case for this config, so there shouldn't be behavior changes. But I still don't full understand whether this config type actually has any influence on the config file we get. It seems to me that it is only used for UI stuff... Anyway I'll put this task as low priority for now. To finish this PR, I still have to add cross-field validation and more test cases.

Reply:
config from REST is a JSON record. Json has boolean type. Not sure if it is an impact

Reply:
I'll add test about this a let you know the result.

Reply:
Both string type and boolean type would work in user's config, I verified that with the modification in travis_correct_json_json.json.